pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -305,7 +305,10 @@ func NewController(kubeClient kubelib.Client, options Options) *Controller {
 			c,
 			c.namespaces,
 			"Namespaces",
-			func(old *v1.Namespace, cur *v1.Namespace, event model.Event) error {
+			func(_ *v1.Namespace, cur *v1.Namespace, event model.Event) error {
+				if cur == nil {
```

Comment:
Not sure i understand clearly, how could cur be nil. AFAIK it is not possible for native k8s client. 

back to this, if namespace can be nil, the any other kinds can be nil, right?

Reply:
IC, in registerHandlers, you have a wrapper handler which will get the latest object by key, if a delete come immediately after update, this could be nil.

Reply:
Whatever this is not the right fix

Reply:
What is the right fix then?

Reply:
Update the wrapperHandler, when `get` returns nil, we mark it Delete event.

Reply:
https://github.com/istio/istio/pull/43962